sites/fast-website/src/app/components/background-design/background-design.ts

```diff
@@ -1,3 +1,261 @@
+            "resize",
+            () => {
+                if (resizeTimeout) {
+                    resizeTimeout = clearTimeout(resizeTimeout);
+                }
+                resizeTimeout = window.setTimeout(() => this.reflowCanvas(), 100);
+            },
+            false
+        );
+
+        const performAnimation = perf => {
+            const frametime = perf - this.prevPerf;
+            this.update(frametime);
+            this.frame = requestAnimationFrame(performAnimation);
+            this.prevPerf = perf;
+        };
+        performAnimation(window.performance.now());
+    }
+
+    setup() {
+        Object.entries(waveData.layers).map(([key, list]) =>
+            list.map(({ from, to, color }, i) => {
+                const waveBlendData = {
+                    path: this.convertPath(from, color),
+                    start: this.convertPath(from),
+                    end: this.convertPath(to),
+                };
+                if (!this.waveData.hasOwnProperty(key)) {
+                    this.waveData[key] = [];
+                }
+                this.waveData[key].push(waveBlendData);
+            })
+        );
+    }
+
+    convertPath(rawPath, color?: number[]): PathData {
+        let M: MoveTo | undefined;
+        let C: BezierCurveTo[] | MoveTo | undefined = rawPath
+            .split(/(?=[MCS])/)
+            .map(d => {
+                let points = d
+                    .slice(1, d.length)
+                    .split(",")
+                    .map(p => ~~p);
+
+                let pairs: number[] = [];
+                if (points.length === 2) {
+                    M = points.map(p => ~~p) as MoveTo;
+                    return;
+                }
+                for (let i = 0; i < points.length; i++) {
+                    pairs.push(~~points[i]);
+                }
+                return pairs as BezierCurveTo | MoveTo;
+            })
+            .filter(Boolean);
+
+        return { C, M, ...(color && { color }) } as PathData;
+    }
+
+    stepPoint(from, to, increments, step) {
+        return from + ((to - from) / increments) * step;
+    }
+
+    generateWave(from, to, cache, inc) {
+        let colorBlend: number = 0;
+        for (let step = 0; step < inc; step++) {
+            colorBlend += 1 / inc;
+            const newWave = {
+                color: from.color.map(
+                    (c, i) => c * colorBlend + (1 - colorBlend) * to.color[i]
+                ),
+                C: from.C.map((c, n) =>
+                    c.map((v, i) => this.stepPoint(v, to.C[n][i], inc, step))
+                ),
+                M: from.M.map((p, i) => this.stepPoint(p, to.M[i], inc, step)),
+            };
+
+            cache.push(newWave);
+        }
+    }
+
+    updateWave(thisWave, startPoints, endPoints, modifier) {
+        // sin wave drives animation
+        let time = this.time.total / 2;
+        let blend = this.easeInOut(1)(Math.sin(time + modifier));
```

Comment:
nit: looks like you could skip the time assignment and just do:
`const blend = this.easeInOut(1)(Math.sin((this.time.total / 2) + modifier));`